haystack/nodes/file_converter/json.py

```diff
@@ -0,0 +1,95 @@
+import json
+from json import JSONDecodeError
+
+from typing import List, Optional, Any, Dict
+
+import logging
+from pathlib import Path
+
+from haystack import Document
+from haystack.nodes.file_converter import BaseConverter
+
+
+logger = logging.getLogger(__name__)
+
+
+class JsonConverter(BaseConverter):
+    """Extracts text from JSON files and casts it into Document objects."""
+
+    outgoing_edges = 1
+
+    def convert(
+        self,
+        file_path: Path,
+        meta: Optional[Dict[str, Any]] = None,
+        remove_numeric_tables: Optional[bool] = None,
+        valid_languages: Optional[List[str]] = None,
+        encoding: Optional[str] = "UTF-8",
+        id_hash_keys: Optional[List[str]] = None,
+    ) -> List[Document]:
+        """
+        Reads a JSON file and converts it into a list of Documents.
+
+        It's primarily a wrapper around `Document.from_dict()` and, as such, acts as the inverse of `Document.to_dict()`.
+
+        It expects one of these formats:
+        - A JSON file with a list of Document dicts.
+        - A JSONL file with every line containing either a Document dict or a list of dicts.
+
+        :param file_path: Path to the JSON file you want to convert.
+        :param meta: Optional dictionary with metadata you want to attach to all resulting documents.
```

Comment:
Just out of curiosity, do you know what the use-case is here to be able to attach the same meta info to all documents? Looking at a few other FileConverters I can see that the meta option isn't always included.

Reply:
Yes, it seems to be a mixed bag. Plus, most of the other converters output one Document which simply gets the meta attached.

I guess one use-case could be to tag all Documents that came through this parse-and-jsonify process. But just thinking out loud. There isn't a particular use-case in mind for this implementation.

Reply:
Yeah I'm not sure, I think we can leave it as is, but make it clear in the docs that this meta info will be added to every Document loaded from the JSON file, which is a bit different behavior from the other converter nodes as you mentioned.